Lombiq.Hosting.Tenants.Management/Controllers/ShellSettingsEditorController.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Lombiq.Hosting.Tenants.Management.Constants;
+using Lombiq.Hosting.Tenants.Management.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.Modules;
+using OrchardCore.Mvc.Core.Utilities;
+using OrchardCore.Tenants.Controllers;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using static OrchardCore.Tenants.Permissions;
+
+namespace Lombiq.Hosting.Tenants.Management.Controllers;
+
+[Feature(FeatureNames.ShellSettingsEditor)]
+public class ShellSettingsEditorController : Controller
+{
+    private readonly IAuthorizationService _authorizationService;
+    private readonly IShellHost _shellHost;
+    private readonly SemaphoreSlim _tenantConfigSemaphore = new(1);
+    private readonly IShellConfigurationSources _tenantConfigSources;
+
+    public ShellSettingsEditorController(
+        IAuthorizationService authorizationService,
+        IShellHost shellHost,
+        IShellConfigurationSources tenantConfigSources)
+    {
+        _authorizationService = authorizationService;
+        _shellHost = shellHost;
+        _tenantConfigSources = tenantConfigSources;
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    public async Task<IActionResult> Edit(ShellSettingsEditorViewModel model)
+    {
+        if (!await _authorizationService.AuthorizeAsync(User, ManageTenants) ||
+            !_shellHost.TryGetSettings(model.TenantId, out var shellSettings))
+        {
+            return NotFound();
+        }
+
+        var settingsDictionary = JsonConvert.DeserializeObject<Dictionary<string, string>>(model.Json);
```

Comment:
Allow submitting empty settings, thus clearing them.

Reply:
While you can submit `{}` now, submitting an empty editor causes an `ArgumentNullException`. This should be handled the same as submitting an empty JSON.